academy/ibc/token-transfer.md

```diff
@@ -0,0 +1,215 @@
+
+How can tokens be transferred between chains and channels?
+
+**Notice:** you can have different channels between two chains, but you cannot transfer the same token across different channels back and forth.
+
+To understand the application logic for a token transfer, first, you have to determine the **source** chain:
+
+![Source chain](images/sourcechain.png)
+
+Then the application logic can be summarized:
+
+![Application logic]
+
+Shortly you will see the corresponding code. Now again have a look at a transfer from **source** to **sink**:
+
+![Source to sink](images/sourcetosink.png)
+
+Above the **source** is chain A. The source channel is **channel-2** and the destination channel is **channel-40**. You see **transfer/channel-...** because the transfer module will bind to a port, which we name transfer. If chain A sends 100 ATOM tokens, chain B will receive 100 ATOM tokens and append the destination prefix **port/channel-id**. So chain B will mint those 100 ATOM tokens as **transfer/channel-40/atoms**. 
+
+If the tokens are sent back from the **same channel** as they were received:
+
+![Sink to source](images/sinktosource.png)
+
+Chain A will "un-escrow" 100 **ATOM tokens**, thus, the prefix will be removed. Chain B will burn **transfer/channel-40/atoms**.
+
+**Notice:** the prefix determines the **source** chain. If the module sends the token from another channel, chain B will be the source chain and chain A will mint new tokens with a prefix instead of un-escrowing ATOM tokens.
+
+![Source sink logic](images/sourcesinklogic.png)
+
+You already know that an application needs to implement the [IBC Module Interface](https://github.com/cosmos/ibc-go/blob/main/modules/core/05-port/types/module.go), so have a look at the [implementation for the token transfer](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/ibc_module.go), e.g. for `OnChanOpenInit`:
+
+```go
+// OnChanOpenInit implements the IBCModule interface
+func (im IBCModule) OnChanOpenInit(
+  ctx sdk.Context,
+  order channeltypes.Order,
+  connectionHops []string,
+  portID string,
+  channelID string,
+  chanCap *capabilitytypes.Capability,
+  counterparty channeltypes.Counterparty,
+  version string,
+) error {
+  if err := ValidateTransferChannelParams(ctx, im.keeper, order, portID, channelID); err != nil {
+    return err
+  }
+
+  if version != types.Version {
+    return sdkerrors.Wrapf(types.ErrInvalidVersion, "got %s, expected %s", version, types.Version)
+  }
+
+  // Claim channel capability passed back by IBC module
+  if err := im.keeper.ClaimCapability(ctx, chanCap, host.ChannelCapabilityPath(portID, channelID)); err != nil {
+    return err
+  }
+
+  return nil
+}
+```
+
+`OnChanOpenAck`, `OnChanOpenConfirm`, `OnChanCloseInit`, and `OnChanCloseConfirm` will do (almost) no checks.
+
+`OnRecvPacket` on the other hand will decode a packet and apply the transfer token application logic:
```

Comment:
To put it like this, makes it seem as if OnRecvPacket is closely related to the channel handshake steps. Might be worth remembering where in the flow these functions appear